lib/active_merchant/billing/gateways/checkout_v2.rb

```diff
@@ -0,0 +1,212 @@
+        post[:card][:expiryMonth] = format(payment_method.month, :two_digits)
+      end
+
+      def add_customer_data(post, options)
+        post[:email] = options[:email] || "unspecified@example.com"
+        if(billing_address = (options[:billing_address] || options[:address]) && post[:card])
+          post[:card][:billingDetails] = {}
+          post[:card][:billingDetails][:address1] = billing_address[:address1]
+          post[:card][:billingDetails][:address2] = billing_address[:address2]
+          post[:card][:billingDetails][:city] = billing_address[:city]
+          post[:card][:billingDetails][:state] = billing_address[:state]
+          post[:card][:billingDetails][:country] = billing_address[:country]
+          post[:card][:billingDetails][:postcode]    = billing_address[:zip]
+          post[:card][:billingDetails][:phone] ={}
+          post[:card][:billingDetails][:phone][:number] = billing_address[:phone]
+        end
+      end
+
+      def add_reference(post, authorization)
+        transaction_id, transaction_amount = split_authorization(authorization)
+        post[:transaction_id] = transaction_id
+        post[:transaction_amount] = transaction_amount
+      end
+
+      def commit(action, post, authorization = nil)
+        begin
+          raw_response = ssl_post(url(post, action, authorization), post.to_json, headers)
+          response = parse(raw_response)
+        rescue ResponseError => e
+          raise unless(e.response.code.to_s =~ /4\d\d/)
+          response = parse(e.response.body)
+        end
+
+          succeeded = success_from(response)
+          Response.new(
+            succeeded,
+            message_from(succeeded, response),
+            response,
+            authorization: authorization_from(response),
+            error_code: error_code_from(succeeded, response),
+            test: test?,
+            avs_result: avs_result(action, response),
+            cvv_result: cvv_result(action, response),
+          )
+      end
+
+      def headers
+        {
+          "Authorization" => @options[:secret_key],
+          "Content-Type"  => "application/json;charset=UTF-8"
+        }
+      end
+
+      def url(post,action, authorization)
+        if action == :authorize
+          live_url + "charges/card"
+        else
+          live_url + "charges/" + authorization + "/"+ action.to_s
+        end
+      end
+
+      def avs_result(action, response)
+        action == :purchase ? AVSResult.new(code: response["card"]["avsCheck"]) : nil
+      end
+
+      def cvv_result(action, response)
+        action == :purchase ? CVVResult.new(response["card"]["cvvCheck"]) : nil
+      end
+
+      def parse(body)
+        return {} unless body
+        JSON.parse(body)
+        rescue JSON::ParserError
+          {
+            "message" => "Invalid response received.",
+            "raw_response" => scrub(body)
+          }
+      end
+
+      def parse_element(response, node)
```

Comment:
I'm not seeing where this method gets called.  Could be another :hocho: candidate.